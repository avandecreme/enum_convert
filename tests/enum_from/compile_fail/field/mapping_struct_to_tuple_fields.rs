use enum_convert::EnumFrom;

enum Source {
    Struct {
        aa: i32,
        bb: i32,
    },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from(Source::Struct)]
    Tuple(
        #[enum_from(Source::Struct.aa)] i32,
        #[enum_from(Source::Struct.bb)] i32,
    ),
}

fn main() {}
