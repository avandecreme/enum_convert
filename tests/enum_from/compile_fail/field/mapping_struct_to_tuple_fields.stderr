error: Unexpected mapping to named field for tuple variant
  --> tests/enum_from/compile_fail/field/mapping_struct_to_tuple_fields.rs:12:36
   |
12 |         #[enum_from(Source::Struct.aa)] i32,
   |                                    ^^
